src/api/app/controllers/source_project_config_controller.rb

```diff
@@ -0,0 +1,55 @@
+class SourceProjectConfigController < SourceController
+  # GET /source/:project/_config
+  def show
+    begin
+      # 'project' can be a local Project in database or a String that's the name of a remote project, or even raise exceptions
+      project = Project.get_by_name(params[:project])
+    rescue Project::ReadAccessError, Project::UnknownObjectError => e
+      render_error status: 404, message: e.message
+      return
+    end
+    config = project.is_a?(String) ? ProjectConfigFile.new(project_name: project) : project.config
```

Comment:
but now that you moved this into its own controller, you could split the function right away. E.g. the first couple of lines create the config object of the given project. So I would extract this and return if config is nil - one codeclimate complaint less and already covered.